src/core/model.js

```diff
@@ -157,28 +160,42 @@ export default class Model extends Object3D {
     defines,
     moduleSettings,
     defaultUniforms,
-    program
+    program,
+    shaderCache
   }) {
-    // Assign default shaders if none are provided
-    if (!vs) {
-      vs = MODULAR_SHADERS.vs;
-    }
-    if (!fs) {
-      fs = MODULAR_SHADERS.fs;
-    }
 
-    // Assign default uniforms (if any default shaders are being used)
-    if (vs === MONOLITHIC_SHADERS.vs || fs === MONOLITHIC_SHADERS.fs) {
-      defaultUniforms = defaultUniforms || MONOLITHIC_SHADERS.defaultUniforms;
-    }
+    this.getModuleUniforms = x => {};
+
+    if (!program) {
+      // Assign default shaders if none are provided
+      if (!vs) {
+        vs = MODULAR_SHADERS.vs;
+      }
+      if (!fs) {
+        fs = MODULAR_SHADERS.fs;
+      }
+
+      // Assign default uniforms (if any default shaders are being used)
+      if (vs === MONOLITHIC_SHADERS.vs || fs === MONOLITHIC_SHADERS.fs) {
+        defaultUniforms = defaultUniforms || MONOLITHIC_SHADERS.defaultUniforms;
```

Comment:
Unrelated, but a good time to do it: We should have a deprecation warning here. We want to phase out the monolithic shaders completely in next release.